### PR TITLE
feat: add auth command side

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: internal/view/requests.go [hunk 1]
```diff
@@ -71,6 +73,27 @@ func PrepareDeleteByKey(table string, key ColumnKey, id string) func(db *gorm.DB
 	}
 }
 
+type Key struct {
+	Key ColumnKey
+	Id  string
+}
+
+func PrepareDeleteByKeys(table string, keys ...Key) func(db *gorm.DB) error {
```
Comment (fgerschwiler) [inline internal/view/requests.go:81]: should be added to the tests

Comment (livio-a) [reply internal/view/requests.go:81]: added

File: internal/user/repository/eventsourcing/eventstore.go [hunk 2]
```diff
@@ -333,31 +337,79 @@ func (es *UserEventstore) UserPasswordByID(ctx context.Context, userID string) (
...
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "EVENT-Fds3s", "user has no password")
+	}
+	if err := crypto.CompareHash(user.Password.SecretCrypto, []byte(old), es.PasswordAlg); err != nil {
+		return nil, caos_errs.ThrowInvalidArgument(nil, "EVENT-s56a3", "invalid password")
+	}
+	return es.changedPassword(ctx, user, new, false)
+}
 
+func (es *UserEventstore) changedPassword(ctx context.Context, user *usr_model.User, password string, onetime bool) (*usr_model.Password, error) {
+	secret, err := crypto.Hash([]byte(password), es.PasswordAlg)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/user/repository/eventsourcing/eventstore.go:408]: added

File: internal/auth/repository/eventsourcing/eventstore/auth_request.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+			}
+			users := make([]model.UserSelection, len(userSessions))
+			for i, session := range userSessions {
+				users[i] = model.UserSelection{
+					UserID:           session.UserID,
+					UserName:         session.UserName,
+					UserSessionState: session.State,
+				}
+			}
+			steps = append(steps, &model.SelectUserStep{Users: users})
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/auth/repository/eventsourcing/eventstore/auth_request.go]: moved a part of it to separate func

File: internal/auth/repository/eventsourcing/eventstore/auth_request.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	}
+	return repo.UserEvents.CheckPassword(ctx, userID, password, request.WithCurrentInfo(info))
+}
+
+func (repo *AuthRequestRepo) VerifyMfaOTP(ctx context.Context, authRequestID, userID string, code string, info *model.BrowserInfo) error {
+	request, err := repo.AuthRequests.GetAuthRequestByID(ctx, authRequestID)
+	if err != nil {
+		return err
+	}
+	if request.UserID != userID {
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/auth/repository/eventsourcing/eventstore/auth_request.go:103]: you can't really skip as `nextsteps` will ensure you met criteria

if you verify otp before verifying the password you will still have to verify it
but afterwards you won't have and mfa verification step as ist already verified for this user agent

**Pull request merged**